internal/blocksync/reactor.go

```diff
@@ -513,8 +513,12 @@ FOR_LOOP:
 				didProcessCh <- struct{}{}
 			}
 
+			firstParts, err := first.MakePartSet(types.BlockPartSizeBytes)
+			if err != nil {
+				break FOR_LOOP
```

Comment:
Given that this would previously have been a panic, should we log in this case? Otherwise we're silently exiting the service routine which seems like it could be hard to debug.